flink-tests/src/test/java/org/apache/flink/test/checkpointing/RescalingITCase.java

```diff
@@ -670,12 +676,15 @@ public void run(SourceContext<Integer> ctx) throws Exception {
 				}
 
 				Thread.sleep(100);
```

Comment:
You only have to make sure that the sources have not finished their processing when a savepoint is taken from them. But you can assure that differently by waiting on a future which is completed when the source has processed all a certain amount of elements.